src/domain/relayer/midnight/midnight_relayer.rs

```diff
@@ -276,13 +301,26 @@ where
     }
 
     async fn get_balance(&self) -> Result<BalanceResponse, RelayerError> {
+        // Get the ledger context from the sync service
+        let context = self.sync_service.get_context();
+
+        // Get the signer to access the wallet seed
+        let signer_model = self
+            .signer_repository
+            .get_by_id(self.relayer.signer_id.clone())
+            .await
+            .map_err(|e| RelayerError::ProviderError(format!("Failed to get signer: {}", e)))?;
+
+        let midnight_signer = MidnightSignerFactory::create_midnight_signer(&signer_model)
```

Comment:
For other relayers we are creating signer and inserting into relayer. Check `create_relayer` methods. We can use same approach for midnight relayers if there is no reason to do it at this level.

Reply:
Good catch!